stdlib/Dates/src/deprecated.jl

```diff
@@ -68,3 +68,11 @@ end
 
 @deprecate argerror(msg::String) ArgumentError(msg) false
 @deprecate argerror() nothing false
+
+# FIXME Dates 2.0: _convert should be replaced by convert
+_convert(::Type{Millisecond}, x::Period) = convert(Millisecond, x)
+@noinline function _convert(::Type{Millisecond},
+                            x::Union{Microsecond, Nanosecond})
+    iszero(rem(tons(x), 10^6)) || Base.depwarn("inexact conversion of Microsecond or Nanosecond to Millisecond is deprecated, use `trunc` or `round`", :_convert)
```

Comment:
`Microsecond` only needs zero remainder against 10^3.

Reply:
`x` is converted to nanoseconds first (`tons`), so the remainder is always computed from nanoseconds